pytest_tests/helpers/test_control.py

```diff
@@ -0,0 +1,80 @@
+import logging
+from functools import wraps
+from time import sleep, time
+
+from _pytest.outcomes import Failed
+from pytest import fail
+
+logger = logging.getLogger("NeoLogger")
+
+
+class expect_not_raises:
+    """
+    Decorator/Context manager check that some action, method or test does not raises exceptions
+
+    Useful to set proper state of failed test cases in allure
+
+    Example:
+        def do_stuff():
+            raise Exception("Fail")
+
+        def test_yellow(): <- this test is marked yellow (Test Defect) in allure
+            do_stuff()
+
+        def test_red(): <- this test is marked red (Failed) in allure
+            with expect_not_raises():
+                do_stuff()
+
+        @expect_not_raises()
```

Comment:
It depends. 

The following test case does not need expect_not_raises, since our expectation is to have exception
```    
def test_raises():
      with pytest.raises(Exception):
              do_exceptional_thing()
```

Now, if we have this, we do
```    
def test_usual_things():
      do_some_usual_preparation_thing() << even if we expect it to not raise any exceptions, it should be TEST_DEFECT if something went wrong (because we didn't checked something there)
      with expect_not_raises():
              do_some_usual_thing_which_we_are_testing() << but this thing is the thing we need to test, so we must treat test as FAILED if something went wrong.
```

In general, I would avoid centralized way to apply expect_not_raises, because we will just hide true TEST_DEFECTs in this case.